src/main/java/org/mule/extension/file/internal/lock/FileChannelPathLock.java

```diff
@@ -72,6 +76,48 @@ public boolean tryLock() {
     }
   }
 
+  /**
+   * Due to the limitations of the {@link FileChannel} class, the only way to provided a waiting lock with a specific
+   * timeout that works for 2 or more concurrent locking operations in the same JVM (i.e. in the same Mule App) is to
+   * perform a busy wait loop for the timeout duration. To avoid numerous requests in the loop, on each iteration
+   * the thread is called to sleep for 1/20th of the duration of the timeout.
+   *
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean tryLock(Long timeout) {
+    long nanoTimeout = timeout < 0 ? 0 : TimeUnit.MILLISECONDS.toNanos(timeout);
+    long startTime = System.nanoTime();
+    do {
+      try {
+        lock = channel.tryLock();
+        if (lock != null) {
+          return isLocked();
+        }
+      } catch (OverlappingFileLockException e) {
```

Comment:
Does this means that we only wait if there is someone else using the lock? Aren't there any other reasons worth waiting?

Reply:
Yes, we only wait if someone else is using the lock. I cannot think of other reasons for waiting / what can cause the tryLock to fail.